server/subscribe.go

```diff
@@ -55,6 +62,14 @@ type ChannelSubscription struct {
 	InstanceID types.ID            `json:"instance_id"`
 }
 
+type SubscriptionTemplate struct {
+	ID         string               `json:"id"`
+	ChannelID  string               `json:"channel_id"`
```

Comment:
I'm thinking subscription templates should be just per-project, and not per-channel. We want users to be able to use the templates across different channels. We'll also only want to expose templates for projects that the user has access to in Jira. Maybe we can lump in the templates in one of the existing API endpoints related to fetching subscriptions, to avoid duplicate project checking

Reply:
@mickmister Sorry I didn't get you, can you explain it a bit. 
Currently we are storing the subscription template with the project Key only. Here is the code related to the [addSubscriptionTemplate](https://github.com/Brightscout/mattermost-plugin-jira/blob/eec7f78830a9313d5f06191fc1b2e477375977fe/server/subscribe.go#L403)

Reply:
@Kshitij-Katiyar What I'm not understanding is why does the template have a `ChannelID` property? The template should be agnostic to which channel it will be used in

Reply:
@mickmister @wiggin77 The channel ID basically used to post a ephemeral in the channel when a template is added/updated/deleted. 
The template is created per-project only and the channel ID has not effect in it.

Reply:
Should that message be posted in all channels where the template is being used?

Reply:
@fmartingr It's an ephemeral message so it is visible only to the user who is updating the template. So, I don't think it should be posted in all the channels.